components/scripts/gradle/03-validate-local-build-caching-different-locations.sh

```diff
@@ -158,8 +170,33 @@ execute_second_build() {
 }
 
 fetch_build_cache_metrics() {
-  read_build_scan_metadata
-  fetch_and_read_build_scan_data build_cache_metrics_only "${build_scan_urls[@]}"
+  if [ "$build_scan_publishing_mode" == "on" ]; then
+    read_build_scan_metadata
+    fetch_and_read_build_scan_data build_cache_metrics_only "${build_scan_urls[@]}"
+  else
+    find_build_scan_dumps
+    read_build_scan_dumps
+  fi
+}
+
+find_build_scan_dumps() {
+  local first_build_dir="${EXP_DIR}/first-build_${project_name}"
+  first_build_scan_dump="$(find "$first_build_dir" -maxdepth 1 -regex '^.*build-scan.*\.scan$')"
```

Comment:
Should we handle the case where this finds more than one file by this pattern?

Reply:
I think so. If we find more than one, then I think it makes sense to use the most recent one. If we find more than one Build Scan dump, then the most likely reason is because one or more Build Scan dumps were committed to the repo. We want the one we just generated, which should be more recent than any that were committed to the repo.

It looks Build Scan dump filenames are sortable (they include the build tool version, the Gradle Enterprise plugin version, and then a timestamp), so we could achieve this by adding `| sort | tail -n 1` to the command above. However, this won't work correctly if the extra Build Scan dumps were generated by a newer version of Gradle or the Gradle Enterprise plugin. That seems rather unlikely to me though.

We could also try modifying the find command itself to return the files sorted by oldest-newest instead, based on the file's last modified time.

Reply:
I just thought of another option. We could compare which files existed before and after the experiment which match the pattern. We know we only call the build script a single time so there should only be a single file. The only edge case I can think of is if the build itself creates a file which matches that pattern. 🤷

What do you think? I could also just fail for now if we detect more than one file and handle this in a separate issue since it's an edge case.

Reply:
I've addressed this with 225a2149591a8f53fd064e489fa11874e128c2de. The simplest approach that should catch the majority of cases is `| sort | tail -n 1`. I tested this by running in interactive mode and generating several Build Scan dumps between the `git clone` and the actual build. Despite there being existing Build Scan dumps, this method chose the most recent.